src/containers/upload-page/store/sagas.js

```diff
@@ -526,10 +527,43 @@ export function* handleUploads({
       console.debug(
         `Attempting to register tracks: ${JSON.stringify(sortedMetadata)}`
       )
-      // Send it off to get our trackIDs
-      const { trackIds, error } = yield AudiusBackend.registerUploadedTracks(
-        sortedMetadata
-      )
+
+      // Send the tracks off to chain to get our trackIDs
+      //
+      // We want to limit the number of concurrent requests to chain here, as tons and tons of
+      // tracks lead to a lot of metadata being colocated on the same block and discovery nodes
+      // can have a hard time keeping up. [see AUD-462]. So, chunk the registration into "rounds."
+      //
+      // Multi-track upload does not have the same issue because we write to chain immediately
+      // after each upload succeeds and those fire on a rolling window.
+      // Realistically, with a higher throughput system, this should be a non-issue.
+      let trackIds = []
+      let error = null
+      for (
+        let i = 0;
+        i < sortedMetadata.length;
+        i += MAX_CONCURRENT_REGISTRATIONS
+      ) {
+        const concurrentMetadata = sortedMetadata.slice(
+          i,
+          i + MAX_CONCURRENT_REGISTRATIONS
+        )
+        const {
```

Comment:
I think there's a small bug here - if we return an error in the current batch, we break the for loop before we append the returned track ids to the outer trackIds slice, which IIRC we need to have for reverting associations later on.